File: Stars Pattern/Right Pascal's Triangle/Pattern.py
#!/usr/bin/env python3

"""
Right Pascal's Triangle Pattern
"""

from sys import path

path.append("../.")


from ConsoleNumInput import inInt


#### Begin ####

n = inInt().getInput(5)

# Method 1
print("Method 1")

for x in range(n): 
    for y in range(x + 1, n):
        print("   ", end="")
        
    for y in range(x + 1):
        print(" * ", end="")
    
    print("")

for x in range(n - 1):
    
    for y in range(x + 1):
        print("   ", end="")
        
    for y in range(x + 1, n):
        print(" * ", end="")
    
    print("")

print("\n*~*~*~*~*~*~*~*~*~*~*~*\n")


# Method 2
print("Method 2")

for x in range(n):
    print("   " * (n - (x + 1)), end="")
    print(" * " * (x + 1))


for x in range(n - 1):
    print("   " * (x + 1), end="")
    print(" * " * (n - x - 1))


print("")
"""
Author: Jovan De Guia
Github Username: jxmked
"""

File: Stars Pattern/Right Pascal's Triangle/Pattern.c
#include <stdio.h>
#include <stdlib.h>

// External File
#include "../ConsoleNumInput.c"

/**
 * Right Pascal's Triangle Pattern
 * */

static int loop(int n);


int main(void){
    // Default
    int n = 5;
    
    do {
        // From External File
        clrscr();
        
        loop(n);
        
        n = inInt();
        
        // This will never trigger. 
        // System required. Hahaha
        if(! n){
            return 0;
        }
    } while(1);
}

 int loop(int n){
    int i, j, r;
    
    /**
     * I just want to make the block of code smaller
     * 
     * */
    for(i = 0; i <= (n * 2); i++){
        
        r = i;
        
        if(i > n){
            r = ((n * 2) - i);
        }
        
        for(j = 0; j <= ((n - r) - 1); j++){
            printf("   ");
        }
        
        for(j = 0; j <= r; j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    
    return 0;
}

/**
 * Written by Jovan De Guia
 * Github Username: jxmked
 * */
